Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/CrashesContentPage.xaml.cs

```diff
@@ -25,27 +25,41 @@ public CrashesContentPage()
             CrashesEnabledSwitchCell.IsEnabled = await MobileCenter.IsEnabledAsync();
         }
 
-        void TestCrash(object sender, EventArgs e)
+        async void UpdateEnabled(object sender, ToggledEventArgs e)
         {
-            Crashes.GenerateTestCrash();
+            await Crashes.SetEnabledAsync(e.Value);
         }
 
-        void DivideByZero(object sender, EventArgs e)
+        void HandleOrThrow(Action action)
         {
-            /* This is supposed to cause a crash, so we don't care that the variable 'x' is never used */
-#pragma warning disable CS0219
-            int x = (42 / int.Parse("0"));
-#pragma warning restore CS0219
+            try
+            {
+                action();
+            }
+            catch (Exception e)
```

Comment:
I think this would be valid C# (and a bit neater):
```
try
{
    action();
}
catch (Exception e) when (HandleExceptionsSwitchCell.On)
{
    Crashes.TrackException(e);
}
```

Reply:
@achocron I like your C# 6 exception filter approach! 👍

Reply:
Will get back to it once develop is fixed for cake.

Reply:
fixed